### PR TITLE
Reddit Link Plugin

### PR DESCRIPTION
**Pull request by olsenej**
Listens for string patterns that match subreddits. Responds to /r/subreddit and r/subreddit. Returns hyperlink to http://reddit.com/r/subreddit.

### CODE REVIEW

File: opt/panicbutt/plugins/reddit.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#  Last Modified : Mon 12 February 2018 10:31:27 AM EST
+#
+#  Created By : Edward Olsen
+#
+# ========================================
+
+import slackbot.bot
+import re
+
+REDDIT_REGEX = "((\/r\/)|(r\/))(\w+)"
```
Comment (Resisty) [inline opt/panicbutt/plugins/reddit.py]: I wonder if we want to pad the token we're looking for with spaces, like
`\s+((\/r\/)|(r\/))(\w+)\s+` to try and reduce false positives/trolling.

Unless that's the point.

**Review by Resisty — CHANGES_REQUESTED** [review]

File: opt/panicbutt/plugins/reddit.py (hunk 1 above)
Comment (olsenej) [reply opt/panicbutt/plugins/reddit.py]: Good catch though I think the pattern \s?+ would work better. There might not always be a space before the match.

Comment (k8r-io) [reply opt/panicbutt/plugins/reddit.py]: Doesn't \s* do what you want?

Comment (olsenej) [reply opt/panicbutt/plugins/reddit.py]: The regex tester I'm using doesn't like the \s*. Seems to throw off the parenthesis in the first group.

**Review by olsenej — COMMENTED** [review]
Latest commit has the suggested change.

File: opt/panicbutt/plugins/reddit.py (hunk 1 above)
Comment (Resisty) [reply opt/panicbutt/plugins/reddit.py]:
```
>>> REDDIT_REGEX = r'((\/r\/)|(r\/))(\w+)'
>>> s = 'find me a subreddit'
>>> t = 'find me r/trees'
>>> r = 'find me buttsr/trees'
>>> q = 'find me lolbutts/r/trees'
>>> for i in [s,t,r,q]:
...     if re.search(REDDIT_REGEX, i, re.I):
...         print('Pattern matches "%s"' % i)
...
Pattern matches "find me r/trees"
Pattern matches "find me buttsr/trees"
Pattern matches "find me lolbutts/r/trees"
>>> NEW_REGEX = r'\s*((\/r\/)|(r\/))(\w+)'
>>> for i in [s,t,r,q]:
...     if re.search(NEW_REGEX, i, re.I):
...         print('Pattern matches "%s"' % i)
...
Pattern matches "find me r/trees"
Pattern matches "find me buttsr/trees"
Pattern matches "find me lolbutts/r/trees"
>>>
>>> OTHER_REGEX = r'\s?+((\/r\/)|(r\/))(\w+)'
>>> for i in [s,t,r,q]:
...     if re.search(NEW_REGEX, i, re.I):
...         print('Pattern matches "%s"' % i)
...
Pattern matches "find me r/trees"
Pattern matches "find me buttsr/trees"
Pattern matches "find me lolbutts/r/trees"
>>>
```

So with `\s*` and `\s?+`, I'm not convinced it's doing what we want; maybe it's acceptable to match `buttsr/trees` or `reddit.com/r/trees` but that wasn't my interpretation of what this plugin was for, which was a short-subreddit expander/linker.

Just my two cents.

**Comment by Resisty** [comment]
[Last thought before merge](https://github.com/Resisty/panicbutt/pull/11/files/57e3491bce49a9d143176dc67321881862dbbb06#r168197188)

**Pull request merged**